### PR TITLE
tools: eksctl to v0.112.0

### LINKED ISSUE #573: unable to test aws-k8s-1.23
I get an error during cluster creation for `aws-k8s-1.23`:
```
❯ kubectl logs -n testsys-bottlerocket-aws x86-64-aws-k8s-123-creation-9gmlt
[2022-09-15T04:52:15Z INFO  resource_agent::agent] Initializing Agent
[2022-09-15T04:52:15Z INFO  eks_resource_agent::eks_provider] Beginning creation of EKS cluster 'x86-64-aws-k8s-123' with creation policy 'Some(IfNotExists)'
[2022-09-15T04:52:15Z INFO  bottlerocket_agents] Getting credentials for assumed role '<REDACTED>'.
[2022-09-15T04:52:16Z INFO  eks_resource_agent::eks_provider] Creation policy is 'IfNotExists' and cluster 'x86-64-aws-k8s-123' does not exist: creating cluster
[2022-09-15T04:52:16Z INFO  eks_resource_agent::eks_provider] Creating cluster with eksctl
2022-09-15 04:52:16 [ℹ]  eksctl version 0.96.0
2022-09-15 04:52:16 [ℹ]  using region us-west-2
Error: invalid version, supported values: 1.19, 1.20, 1.21, 1.22
Provider error: An error occurred but no resources were left behind, Failed create cluster with status code exit status: 1
```

Pretty sure this just boils down to updating `eksctl` to [0.109.0](https://github.com/weaveworks/eksctl/releases/tag/v0.109.0) or later.

### PR DESCRIPTION
**Pull request by ecpullen**
**Issue number:**

Closes #573

**Description of changes:**

Updates eksctl to v0.112.0 which enables configuring iamidentitymappings through a config file.

**Testing done:**

`make tools`
I will also be testing this with #447.
I have verified through #447 that we can create clusters with the updated tools and that the new version of `eksctl` is used.
**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: tools/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,17 @@ The format is inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0
 Since this project is only a vessel for packaging a few binary tools, its adherence to
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html) is loose at best.
 
+## [0.2.0] - 2022-09-20
+
+Update eksctl and sonobuoy.
+
+### Contents
+
+- boringtun v0.4.0
```
Comment (etungsten) [inline tools/CHANGELOG.md]: I thought I had already removed this.

Comment (ecpullen) [reply tools/CHANGELOG.md]: I'll check. The change log hadn't been changed since it was added so I just changed the eksctl bit.

Comment (ecpullen) [reply tools/CHANGELOG.md]: Borington is removed. I'll fix the log

**Review by etungsten — APPROVED** [review]

**Review by mjsterckx — APPROVED** [review]
LGTM

**Review by stmcginnis — APPROVED** [review]
Looks good!

**Pull request merged**